### PR TITLE
[KOGITO-1114] Jobs service tests with Kogito app

### PR DESCRIPTION
**Pull request by sutaakar**
https://issues.redhat.com/browse/KOGITO-1114

Signed-off-by: Karel Suta <ksuta@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: test/features/cr/deploy_quarkus_service.feature [hunk 1]
```diff
@@ -56,4 +56,34 @@ Feature: CR: Deploy quarkus service
     @native
     Examples:
       | native | minutes |
-      | "enabled" | 20 |
\ No newline at end of file
+      | "enabled" | 20 |
+
+#####
+
+  # Disabled as long as https://issues.redhat.com/browse/KOGITO-1163 and https://issues.redhat.com/browse/KOGITO-1166 is not solved
```
Comment (radtriste) [inline test/features/cr/deploy_quarkus_service.feature]: Do we have a follow-up task on those issues to enable the test when they are solved ? :)

File: test/features/cr/deploy_quarkus_service.feature [hunk 2]
```diff
@@ -56,4 +56,34 @@ Feature: CR: Deploy quarkus service
...
+
+    # Implement retrieving of job information from Jobs service once https://issues.redhat.com/browse/KOGITO-1163 is fixed
+    Then Kogito application "timer-quarkus-example" log contains text "Before timer" within 1 minutes
+    And Kogito application "timer-quarkus-example" log contains text "After timer" within 1 minutes
+
+    Examples: Non Native
+      | native | minutes |
+      | "disabled" | 10 |
+
+    # Disabled as long as https://issues.redhat.com/browse/KOGITO-1179 is not solved
```
Comment (radtriste) [inline test/features/cr/deploy_quarkus_service.feature]: same here, do we have a follow-up task on this issue to enable the test when it is solved ? :)

File: test/framework/kubernetes.go [hunk 3]
```diff
@@ -133,3 +134,31 @@ func loadResource(namespace, uri string, resourceRef meta.ResourceObject, before
...
+
+		// Container name is equal to deployment config name
+		return checkPodsContainingTextInLog(namespace, pods, dcName, logText)
+	})
+}
+
+func checkPodsContainingTextInLog(namespace string, pods *corev1.PodList, containerName, text string) (bool, error) {
+	for _, pod := range pods.Items {
+		containsText, err := isPodContainingTextInLog(namespace, &pod, containerName, text)
+		if err != nil || !containsText {
```
Comment (radtriste) [inline test/framework/kubernetes.go]: small mistake here I think
you should stop looking into pods if you find something or ?

Comment (sutaakar) [reply test/framework/kubernetes.go]: Actually not. The purpose is to check all pods than they contain required text.
If test is found then another pod is inspected (for cycle continue), if text is not found then false is returned from the method.

File: test/features/cr/deploy_quarkus_service.feature (hunk 1 above)
Comment (sutaakar) [reply test/features/cr/deploy_quarkus_service.feature]: Not yet, will create one.

Comment (sutaakar) [reply test/features/cr/deploy_quarkus_service.feature]: Created https://issues.redhat.com/browse/KOGITO-1265

File: test/framework/kubernetes.go (hunk 3 above)
Comment (radtriste) [reply test/framework/kubernetes.go]: Can we then call the method `*AllPods*` ? would be clearer I think

Comment (sutaakar) [reply test/framework/kubernetes.go]: Done

**Pull request merged**